docs/gen-ai/gen-ai-spans.md

```diff
@@ -172,6 +175,23 @@ Additional output format details may be recorded in the future in the `gen_ai.ou
 <!-- markdownlint-capture -->
 <!-- markdownlint-disable -->
 
+**Status:** ![Development](https://img.shields.io/badge/-development-blue)
+
+Describes tool execution span.
+
+`gen_ai.operation.name` SHOULD be `execute_tool`.
+
+**Span name** SHOULD be `execute_tool {gen_ai.tool.name}`.
+
+GenAI instrumentations that are able to instrument tool execution call SHOULD do so.
+However, it's common for tools to be executed by the application code. It's recommended
+for the application developers to follow this semantic conventions for tool invoked
+by the application code.
+
+**Span kind** SHOULD be `INTERNAL`.
```

Comment:
are these out of proc tools?

Reply:
can be in-proc or out-of-proc

Reply:
It's essentially a function call which may involve a network one.

E.g. I have a `get_weather` tool that retrieves a weather from a remote service, but may also have a local cache. 
Or a tool that asks user for additional input.

What instrumentation has is the reference to the function to call and it can wrap the execution into the span, but it won't know what happens underneath.

Plus, the network operations inside the function call would be traced (if instrumented) and may have client (logical) -> client (network)  layers already. So I'd rather keep `internal` kind.

PS: I almost wish we had a `¯\_(ツ)_/¯` kind instead of `INTERNAL`

Reply:
> the network operations inside the function call would be traced (if instrumented)

👍